include/multipass/timer.h

```diff
@@ -50,14 +52,26 @@ class Timer
 
 private:
     void main();
-    const std::chrono::seconds timeout;
+    const std::chrono::milliseconds timeout;
     const std::function<void()> callback;
     TimerState current_state;
     std::thread t;
     std::condition_variable cv;
     std::mutex cv_m;
 };
 
+#define MP_TIMER_SYNC_FUNCS multipass::utils::TimerSyncFuncs::instance()
+
+class TimerSyncFuncs : public Singleton<TimerSyncFuncs>
+{
+public:
+    TimerSyncFuncs(const Singleton<TimerSyncFuncs>::PrivatePass&) noexcept;
+
+    virtual void notify_all(std::condition_variable& cv);
+    virtual void wait(std::condition_variable& cv, std::unique_lock<std::mutex>& lock);
+    virtual std::cv_status wait_for(std::condition_variable& cv, std::unique_lock<std::mutex>& lock,
+                                    const std::chrono::duration<int, std::milli>& rel_time);
```

Comment:
One more thing I just remembered while looking at other singleton code: these ought to be `const`.

Reply:
Oh, you're absolutely right!  Thanks for pointing that out!

Reply:
Done